ui/page/dcrdex/dex_onboarding_page.go

```diff
@@ -1009,12 +1059,21 @@ func (pg *DEXOnboarding) setAddServerStep() {
 }
 
 func (pg *DEXOnboarding) connectServerAndPrepareForBonding() {
+	dexClient := pg.AssetsManager.DexClient()
 	var xc *core.Exchange
 	var err error
 	if pg.existingDEXServer != "" { // Already registered just want to post bond.
-		xc, err = pg.dexc.Exchange(pg.bondServer.url)
+		xc, err = dexClient.Exchange(pg.bondServer.url)
+	} else if dexClient.InitializedWithPassword() {
+		var paid bool
+		xc, paid, err = dexClient.DiscoverAccount(pg.bondServer.url, pg.dexPass, pg.bondServer.cert)
+		canTrade := paid && xc.ConnectionStatus == comms.Connected && (xc.Auth.EffectiveTier > 0 || len(xc.Auth.PendingBonds) > 0)
```

Comment:
Why do we care about the connection status? Do we even care about paid? Shouldn't we only be bothered about EffectiveTier? If there are pending bonds, we want to display the bond confirmation progress?

Reply:
Ah, right.